reactor-core/src/test/java/reactor/core/publisher/OptimisticEmitFailureHandlerTest.java

```diff
@@ -0,0 +1,36 @@
+package reactor.core.publisher;
+
+import java.time.Duration;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+class OptimisticEmitFailureHandlerTest {
+
+	@Test
+	void shouldRetryOptimistically() {
```

Comment:
can you add a similar test that _does_ validate that the FAIL_NON_SERIALIZED comes through if the deadline is passed ?